### PR TITLE
Adding VIM API for sw-patch

### PR DESCRIPTION
**Pull request by enzocandotti**
Adding VIM API for sw-patch
    
This commit introduces a VIM API in order to let DM to create, apply,
abort and delete sw-patch strategies.

The idea of these changes is to add the structure of the VIM API. The
sw-patch URL may be changed on later commits, depending on the
strategy needed to handle from DM.

This also adds unit tests for sw-patch API calls.

Signed-off-by: Enzo Candotti <enzo.candotti@windriver.com>

### CODE REVIEW

File: starlingx/nfv/orchestration/doc.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/* Copyright(c) 2019 Wind River Systems, Inc. */
+
+package orchestration
```
Comment (yjian118) [inline starlingx/nfv/orchestration/doc.go]: Considering use version v1 of the nfv API. The orchestration may not be necessary.

File: starlingx/nfv/orchestration/swpatch/requests.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/* Copyright(c) 2023 Wind River Systems, Inc. */
+
+package swpatch
+
+import (
+	"encoding/json"
+	"github.com/gophercloud/gophercloud"
+	inventoryv1 "github.com/gophercloud/gophercloud/starlingx/inventory/v1"
```
Comment (yjian118) [inline starlingx/nfv/orchestration/swpatch/requests.go]: Can we move the common utilities out of the inventory and nfv package?

File: starlingx/nfv/orchestration/doc.go (hunk 1 above)
Comment (enzocandotti) [reply starlingx/nfv/orchestration/doc.go]: Done.
Updated to v1.

File: starlingx/nfv/orchestration/swpatch/requests.go (hunk 2 above)
Comment (enzocandotti) [reply starlingx/nfv/orchestration/swpatch/requests.go]: I've moved the patch.go file from github.com/gophercloud/gophercloud/starlingx/inventory/v1 to github.com/gophercloud/gophercloud/starlingx/, and renamed all the imports to use it as "common".

**Comment by enzocandotti** [comment]
Unit tests added.
I will proceed to create a client on DM to use this API.

**Comment by enzocandotti** [comment]
In [starlingx/client.go](https://github.com/Wind-River/gophercloud/blob/starlingx/starlingx/client.go), we have:
`
func initClientOpts(client *gophercloud.ProviderClient, eo gophercloud.EndpointOpts, clientType string)(*gophercloud.ServiceClient, error) {
	sc := new(gophercloud.ServiceClient)
	eo.ApplyDefaults(clientType)
	url, err := client.EndpointLocator(eo)
	if err != nil {
		return sc, err
	}
	sc.ProviderClient = client
	sc.Endpoint = url
	sc.Type = clientType
	return sc, nil
}

// NewInventoryV1 creates a ServiceClient that may be used with the v1 inventory package.
func NewInventoryV1(client *gophercloud.ProviderClient, eo gophercloud.EndpointOpts) (*gophercloud.ServiceClient, error) {
	return initClientOpts(client, eo, "inventory")
}


I'm wondering if we have to do something similar here. But IDK whether this "NewInventoryV1" function is being used, since I don't see any calls from DM side.

**Review by yjian118 — COMMENTED** [review]
May need to update the copyrights on all the affected files, others looks good to me

**Review by yjian118 — APPROVED** [review]

**Comment by enzocandotti** [comment]
> May need to update the copyrights on all the affected files, others looks good to me

Done. Also updated all the occurrences in order to refer to 'sw-patch'

**Review by ttakenak — APPROVED** [review]

**Pull request merged**